### PR TITLE
refactor: validation for writing auth models

### PR DESCRIPTION
**Pull request by craigpastro**
## Description

Simplify the validation for auth models. Previously the code was a bit hard to understand... and I was the one who wrote it.

This will make it much easier to update to the new version and also to add the checks on relation types.

## Review Checklist
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: server/commands/write_authzmodel.go [hunk 1]
```diff
@@ -50,115 +55,104 @@ func (w *WriteAuthorizationModelCommand) Execute(ctx context.Context, req *openf
...
+	return nil
+}
 
-		targets, err := validateTypeDefinition(td)
-		if err != nil {
-			return err
-		}
-		tupleToUsersetTargets = append(tupleToUsersetTargets, targets...)
-	}
+func containsDuplicateType(tds []*openfgapb.TypeDefinition) bool {
```
Comment (miparnisari) [inline server/commands/write_authzmodel.go:70]:
```suggestion
func containsDuplicateTypes(tds []*openfgapb.TypeDefinition) bool {
```

File: server/errors/errors.go [hunk 2]
```diff
@@ -84,8 +84,8 @@ func RelationNotFound(relation string, typeName string, tuple *openfgapb.TupleKe
 	return status.Error(codes.Code(openfgapb.ErrorCode_relation_not_found), msg)
 }
 
-func EmptyRelationDefinition(typeDefinition, relation string) error {
-	return status.Error(codes.Code(openfgapb.ErrorCode_empty_relation_definition), fmt.Sprintf("Type definition '%s' contains an empty relation definition '%s'", typeDefinition, relation))
```
Comment (miparnisari) [inline server/errors/errors.go:88]: This is a user-facing error. I feel like the existing message is clearer. https://auth0.slack.com/archives/C033LQD4S11/p1662068452184499?thread_ts=1662066590.039589&cid=C033LQD4S11

File: server/commands/write_authzmodel.go (hunk 1 above)
Comment (craigpastro) [reply server/commands/write_authzmodel.go:70]: Fixed. Thank you!

File: server/errors/errors.go (hunk 2 above)
Comment (craigpastro) [reply server/errors/errors.go:88]: How about now? Previously it was kind of wrong. Do you have any suggestions? It is in the case a user has something like
```
type doc
  relations
    define owner  // and nothing else
```

Comment (miparnisari) [reply server/errors/errors.go:88]: My nitpick is that we don't mention the word `rewrite` anywhere in our documentation.

How about `the definition of relation R on type T is invalid` ?

Comment (craigpastro) [reply server/errors/errors.go:88]: Oh yeah, that's much better. Thanks! Updated.

**Review by miparnisari — APPROVED** [review]

**Pull request merged**